packages/suite/src/components/wallet/PrivacyAccount/CoinjoinSetup.tsx

```diff
@@ -0,0 +1,106 @@
+import styled from 'styled-components';
+
+import { Translation } from '@suite-components';
+import { useSelector } from '@suite-hooks/useSelector';
+import { Card, RadioButton, motionAnimation, Warning } from '@trezor/components';
+import { coinjoinAccountUpdateSetupOption } from '@wallet-actions/coinjoinAccountActions';
+import { AnonymityLevelSetup } from '@wallet-components/PrivacyAccount/AnonymityLevelSetup';
+import { MaxMiningFeeSetup } from '@wallet-components/PrivacyAccount/MaxMiningFeeSetup';
+import { SkipRoundsSetup } from './SkipRoundsSetup';
+import { selectCoinjoinAccountByKey } from '@wallet-reducers/coinjoinReducer';
+
+const SetupContainer = styled.div`
+    padding: 18px;
+`;
+
+const Wrapper = styled.div`
+    display: flex;
+    flex-direction: column;
+`;
+
+const SetupOptions = styled(Wrapper)`
+    align-items: flex-start;
+    gap: 24px;
+`;
+
+const CustomSetup = styled(Wrapper)`
+    border-top: 1px solid ${props => props.theme.STROKE_GREY};
+    gap: 32px;
+    margin-top: 24px;
+    padding-top: 16px;
+`;
+
+interface CoinjoinSetupProps {
+    accountKey: string;
+}
+
+export const CoinjoinSetup = ({ accountKey }: CoinjoinSetupProps) => {
+    const coinjoinAccount = useSelector(state => selectCoinjoinAccountByKey(state, accountKey));
+
+    const dispatch = useDispatch();
+
+    if (!coinjoinAccount) {
+        return null;
+    }
+
+    const hasSession = !!coinjoinAccount.session;
+
+    const handleSetupOptionChange = (isRecommended: boolean) => {
+        if (!!coinjoinAccount.customSetup === isRecommended) {
+            dispatch(coinjoinAccountUpdateSetupOption(accountKey, isRecommended));
+        }
+    };
+    const setRecommendedSetup = () => handleSetupOptionChange(true);
+    const setCustomSetup = () => handleSetupOptionChange(false);
+
+    return (
+        <Card customPadding="8px">
+            {hasSession && (
+                <Warning variant="info">
+                    <Translation id="TR_DISABLED_ANONYMITY_CHANGE_MESSAGE" />
+                </Warning>
+            )}
+            <SetupContainer>
+                <SetupOptions>
+                    <RadioButton
+                        isChecked={!coinjoinAccount.customSetup}
+                        onClick={setRecommendedSetup}
+                        disabled={hasSession}
+                    >
+                        <Translation id="TR_RECOMMENDED" />
+                    </RadioButton>
+                    <RadioButton
+                        isChecked={coinjoinAccount.customSetup}
+                        onClick={setCustomSetup}
+                        disabled={hasSession}
+                    >
+                        <Translation id="TR_CUSTOM" />
+                    </RadioButton>
+                </SetupOptions>
+                <AnimatePresence initial={!coinjoinAccount.customSetup}>
+                    {coinjoinAccount.customSetup && (
+                        <motion.div {...motionAnimation.expand}>
```

Comment:
Add `transition={{ duration: 0.4, ease: motionEasing.transition }}` for smooothness :)

Reply:
c6480024375680650bf1289688c906efe68254ca Smooth indeed.